clang/lib/Sema/SemaDecl.cpp

```diff
@@ -17883,6 +17889,23 @@ Sema::FunctionEmissionStatus Sema::getEmissionStatus(FunctionDecl *FD) {
     // be emitted, because (say) the definition could include "inline".
     FunctionDecl *Def = FD->getDefinition();
 
+    if (Def &&
+        !isDiscardableGVALinkage(getASTContext().GetGVALinkageForFunction(Def)))
+      return FunctionEmissionStatus::Emitted;
+  }
```

Comment:
Is everything ok with this code? I can't find the opening curly brace (`{`) for this one `}`.

Reply:
Fixed